### PR TITLE
Add ScottoMacrodeck Macropad

### PR DESCRIPTION
**Pull request by joe-scotto**
Add ScottoMacrodeck handwired macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+    "manufacturer": "ScottoKeebs",
+    "keyboard_name": "ScottoMacrodeck",
+    "maintainer": "joe-scotto",
+    "bootloader": "caterina",
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/info.json:5]:
```suggestion
    "development_board": "promicro",
```
This specifies that it's an atmega32u4, and uses the caterina bootloader (as well as being pro micro pin compatible, for converters)

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "nkro": true
+    },
+    "matrix_pins": {
+        "direct": [
+            ["D3", "D2", "D1", null],
+            ["D0", "D4", "C6", "D7"],
+            ["E6", "B4", "B5", "F4"]
+        ]
+    },
+    "processor": "atmega32u4",
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/info.json:21]:
```suggestion
```

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "bootloader": "caterina",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/info.json:14]:
```suggestion
    "build" : {
        "lto": true,
    },
    "matrix_pins": {
```

File: keyboards/handwired/scottokeebs/scottomacrodeck/rules.mk [hunk 4]
```diff
@@ -0,0 +1 @@
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/rules.mk:1]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        ]
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x534B"
+    },
+    "layouts": {
+        "LAYOUT_ortho_3x4": {
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/info.json:31]: This isn't quite ortho 3x4.  It's missing a key for that. 
```suggestion
        "LAYOUT": {
```

File: keyboards/handwired/scottokeebs/scottomacrodeck/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_3x4(
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/keymaps/default/keymap.c:21]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/handwired/scottokeebs/scottomacrodeck/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_3x4(
```
Comment (drashna) [inline keyboards/handwired/scottokeebs/scottomacrodeck/keymaps/via/keymap.c:21]:
```suggestion
    [0] = LAYOUT(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/scottokeebs/scottomacrodeck/readme.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+# Compiling
+
+Make example for this keyboard (after setting up your build environment):
+
+    make handwired/jscotto/scottomacrodeck:default
+
+Flashing example for this keyboard:
+
+    make handwired/jscotto/scottomacrodeck:default
```
Comment (fauxpark) [inline keyboards/handwired/scottokeebs/scottomacrodeck/readme.md:19]:
```suggestion
    make handwired/scottokeebs/scottomacrodeck:default

Flashing example for this keyboard:

    make handwired/scottokeebs/scottomacrodeck:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/scottokeebs/scottomacrodeck/readme.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# ScottoMacrodeck
+
+![ScottoMacrodeck](https://imgur.com/KVJuIVNh.jpeg)
+
+An 11 key VIA compatible macro pad. Case files available [here](https://github.com/joe-scotto/scottokeebs).
+
+-   Keyboard Maintainer: [Joe Scotto](https://github.com/joe-scotto)
+-   Hardware Supported: ATmega32U4
+-   Hardware Availability: [Amazon](https://amazon.com)
```
Comment (fauxpark) [inline keyboards/handwired/scottokeebs/scottomacrodeck/readme.md:9]:
```suggestion
* Keyboard Maintainer: [Joe Scotto](https://github.com/joe-scotto)
* Hardware Supported: ATmega32U4
* Hardware Availability: [Amazon](https://amazon.com)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json (hunk 5 above)
Comment (fauxpark) [reply keyboards/handwired/scottokeebs/scottomacrodeck/info.json:31]: Still outstanding.

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please revert the submodule changes.

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            ["D3", "D2", "D1", null],
+            ["D0", "D4", "C6", "D7"],
+            ["E6", "B4", "B5", "F4"]
+        ]
+    },
+    "url": "https://scottokeebs.com",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0015",
+        "vid": "0xFEED"
```
Comment (fauxpark) [inline keyboards/handwired/scottokeebs/scottomacrodeck/info.json:28]: Vendor ID cannot be FEED for VIA support.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/scottokeebs/scottomacrodeck/info.json (hunk 10 above)
Comment (joe-scotto) [reply keyboards/handwired/scottokeebs/scottomacrodeck/info.json:28]: Resolved by [7f4ae5e](https://github.com/qmk/qmk_firmware/pull/20946/commits/7f4ae5e66ce1e3b4e9d4865cc01145953ee9c1fa). The board doesn't need via, I'm hosting Vial on my website instead for GUI support.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**